File: DsatPsatLmsApi/Models/TestSession.cs
using System;
using System.Collections.Generic;

namespace DsatPsatLmsApi.Models
{
    public class TestSession
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public User User { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime? EndTime { get; set; }
        public string ModuleRoute { get; set; } = "Medium"; // Low, Medium, High
        public int TotalQuestions { get; set; }
        public int CorrectAnswers { get; set; }
        public int Score { get; set; }
        public List<SessionQuestion> Questions { get; set; } = new List<SessionQuestion>();
    }
}

File: server/src/models/TestSession.js
import mongoose from 'mongoose'

const questionSchema = new mongoose.Schema({
  content: String,
  subject: String,
  difficulty: String,
  correctAnswer: String,
  userAnswer: String,
  timeSpent: Number
})

const schema = new mongoose.Schema({
  userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', index: true },
  moduleRoute: String,
  startTime: { type: Date, default: Date.now },
  endTime: Date,
  totalQuestions: Number,
  correctAnswers: Number,
  score: Number,
  questions: [questionSchema]
})

export default mongoose.model('TestSession', schema)
